src/maicos/core/base.py

```diff
@@ -451,47 +451,69 @@ def _call_single_frame(self, ts, current_frame_index) -> None:
 
         self._obs = Results()
+        self._var = Results()
+        self._pop = Results()
 
         self.timeseries[current_frame_index] = self._single_frame()
 
         # This try/except block is used because it will fail only once and is
         # therefore not a performance issue like a if statement would be.
         try:
+            # Fail fast if the means and sems are not defined yet.
+            self.means  # type: ignore  # noqa B018
+            self.sems  # type: ignore  # noqa B018
+
             for key in self._obs:
-                if type(self._obs[key]) is list:
+                # Sanitize the data type of the observable
+                if isinstance(self._obs[key], list):
                     self._obs[key] = np.array(self._obs[key])
-                old_mean = self.means[key]  # type: ignore
-                old_var = self.sems[key] ** 2 * (self._index - 1)  # type: ignore
-                self.means[key] = new_mean(  # type: ignore
+                if key not in self._pop:
+                    self._pop[key] = np.ones(np.shape(self._obs[key]), dtype=int)
+                    self._var[key] = np.zeros(np.shape(self._obs[key]), dtype=float)
+
+                self.pop[key], self.means[key], self.M2[key] = parallel_welford(  # type: ignore
+                    self._pop[key],  # type: ignore
+                    self.pop[key],  # type: ignore
+                    self._obs[key],  # type: ignore
                     self.means[key],  # type: ignore
-                    self._obs[key],
-                    self._index,  # type: ignore
-                )  # type: ignore
-                self.sems[key] = np.sqrt(  # type: ignore
-                    new_variance(
-                        old_var,
-                        old_mean,
-                        self.means[key],  # type: ignore
-                        self._obs[key],
-                        self._index,
-                    )
-                    / self._index
+                    self._var[key] * self._pop[key],  # type: ignore
+                    self.M2[key],  # type: ignore
                 )
-                self.sums[key] += self._obs[key]  # type: ignore
+
+                self.sems[key] = np.sqrt(self.M2[key] / self.pop[key] ** 2)  # type: ignore
+                self.sums[key] += self._obs[key] * self._pop[key]  # type: ignore
 
         except AttributeError as err:
             with logging_redirect_tqdm():
                 logging.debug("Initializing error estimation.")
             # the means and sems are not yet defined. We initialize the means with
             # the data from the first frame and set the sems to zero (with the
             # correct shape).
-            self.sums = self._obs.copy()
-            self.means = self._obs.copy()
+            self.sums = Results()
+            self.means = Results()
             self.sems = Results()
+            self.pop = Results()
+            self.M2 = Results()
```

Comment:
Do we have a developer dev to explain that these attributes exist and how to use them?

Reply:
We don't, but we should. I will write it down in the dev documentation

Reply:
Thanks, [own_module.rst](https://github.com/maicos-devel/maicos/blob/main/docs/src/devdoc/own_module.rst) seems to be the dedicated spot.  In fact this doc page could benefit from some love...